yb-voyager/cmd/analyzeSchema.go

```diff
@@ -197,6 +198,30 @@ func reportSummary() {
 		reportStruct.Summary.DBObjects = append(reportStruct.Summary.DBObjects, dbObject)
 	}
 }
+// Checks Whether there is a GIN index
+func checkGin(sqlInfoArr []sqlInfo, fpath string) {
+	for _, sqlInfo := range sqlInfoArr {
+		idx := ginRegex.FindStringSubmatch(sqlInfo.stmt)
+		if idx != nil {
+			columnsFromGin :=  strings.Trim(idx[4], `()`)
+			columnList := strings.Split(columnsFromGin, ",")
+			if len(columnList) > 1 {
+				reportCase(fpath, "Schema contains gin index on multi column which is not supported.",
+					"https://github.com/yugabyte/yugabyte-db/issues/7850", "", "INDEX", idx[2], sqlInfo.formattedStmtStr)
+			} else {
+				if strings.Contains(strings.ToUpper(columnList[0]), "ASC") || strings.Contains(strings.ToUpper(columnList[0]), "DESC") || strings.Contains(strings.ToUpper(columnList[0]), "HASH") {
+					reportCase(fpath, "Schema contains gin index on column with ASC/DESC/HASH Clause which is not supported.",
+						"https://github.com/yugabyte/yugabyte-db/issues/7850", "", "INDEX", idx[2], sqlInfo.formattedStmtStr)
+				}
+			}
+		}
+		if strings.Contains(strings.ToLower(sqlInfo.stmt), "extension btree_gin") {
+			reportCase(fpath, "Schema contains btree_gin extension which is not supported for YB",
```

Comment:
Can you define a local anonymous function to factor out common values:

```
reportFn := function(msg string, ghIssueID int, name string) {
    reportCase(fpath, msg, fmt.Sprintf(""https://github.com/yugabyte/yugabyte-db/issues/%d", ghIssueID), "INDEX", name, sqlInfo.formattedStmtStr)
}
```

Reply:
@amit-yb should this be considered in the refactoring of analyzeSchema later?